macosx/QuickLookExtension/PreviewViewController.mm

```diff
@@ -0,0 +1,146 @@
+// This file Copyright © 2010-2022 Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+#import "PreviewViewController.h"
+
+#include <libtransmission/error.h>
+#include <libtransmission/torrent-metainfo.h>
+
+#import "NSStringAdditions.h"
+
+#import <QuickLookUI/QLPreviewingController.h>
+
+#pragma mark - TRError
+
+@interface TRError : NSError
+
+- (instancetype)initWith:(tr_error*)error;
+
+@end
+
+@interface TRError ()
```

Comment:
I am not sure what problem this class solves. We can return the plain NSError supplied with the right parameters. What does an extra class achieve?

Reply:
This was more like a little bit of my playground here.
NSError can be `init`ed with error code and domain, but to provide message (which we already have as `char*` in `tr_error`) you need to subclass.
See [Subclassing notes for NSError](https://developer.apple.com/documentation/foundation/nserror#:~:text=for%20more%20information.-,Subclassing%20Notes,-Applications%20may%20choose)

Yeah, I won't include this in final impl.
Maybe it's good idea to create this NSError<->tr_error wrapper in common place for all macosx related code.

Reply:
You don't actually need to subclass though. Localized description is stored in the user info dictionary and can be retrieved from any NSError. See [localizedDescription docs](https://developer.apple.com/documentation/foundation/nserror/1414418-localizeddescription?language=objc)

Reply:
Ohh, thanks!